### ISSUE TITLE
Skipped because proportional CJK

### ISSUE THREAD

**Issue by yisibl** [kojiishi/east_asian_spacing#175]
I am trying to execute a command on a Chinese font:

```
ast-asian-spacing -o build foo.ttf

Skipped because proportional CJK: "foo font name"
```

Does this project have any special requirements for the glyphs in the font or what characters need to be included?

**Comment by kojiishi** [comment]
> Does this project have any special requirements for the glyphs in the font or what characters need to be included?

Yes, because the CJK em-box isn't well-defined in the OpenType format, it needs a heuristic to determine it. The current list is chosen to see if two different fullwidth punctuation characters have the same width as 3 commonly used ideographic characters. Please see:
https://github.com/kojiishi/east_asian_spacing/blob/main/east_asian_spacing/config.py#L41

If your font doesn't have all these glyphs, the current logic fails. I could add some fallback logic, but I wasn't sure what options/fallback logic to add, so I relied the users to provide one.

You can specify the CJK em-box size or characters to compute it from by the `--em` option:
https://github.com/kojiishi/east_asian_spacing/blob/main/east_asian_spacing/builder.py#L229
or you can create a python code to customize it like this:
https://github.com/googlefonts/chws_tool/blob/main/src/chws_tool/config.py#L63

I have to admit that the former isn't well tested, please let me know if it doesn't work as you expect.

**Comment by yisibl** [comment]
@kojiishi Thanks for the answer, indeed, the width of `「」（）` in my test font is less than em-box (1000).

**Comment by yisibl** [comment]
I have to say this is a very great project!

Is it possible to add an option to generate fea files for a specific language? That way I can manually compile chws into fonts via fonttools, which is sometimes more flexible.

**Comment by yisibl** [comment]
This is my [test font](https://iconfont.alicdn.com/wf/font_source/P1EXxgPTCEW1/ZojminmzNTtQ.zip) (free for commercial use):

**Comment by kojiishi** [comment]
> @kojiishi Thanks for the answer, indeed, the width of `「」（）` in my test font is less than em-box (1000).

If all punctuation characters have the same widths, you should be able to use `--em '「」（）'` or `--em 900`. Isn't it the case?
